### PR TITLE
[qt5]Fix static build failure in windows

### PR DESCRIPTION
**Pull request by JackBoosY**
1. The option `staticlib` is no longer supported in static builds, replacing it with `static`.
2. Now we don't need to use `qmake_all` to build qt5 and other modules in windows-static, it seems to **break** the disabled parameters during the build. See [documentation](https://wiki.qt.io/Building_Qt_Documentation).

Fix the failure of qt5 module build in windows-static (caused by #5293):

- [x] qt5-base
- [x] qt5-3d
- [x] qt5-activeqt : only support Windows
- [x] qt5-charts
- [x] qt5-connectivity
- [x] qt5-datavis3d
- [x] qt5-declarative
- [x] qt5-gamepad
- [x] qt5-graphicaleffects
- [x] qt5-imageformats
- [x] qt5-location
- [x] qt5-macextras : only support OSX
- [x] qt5-modularscripts
- [x] qt5-mqtt
- [x] qt5-multimedia
- [x] qt5-networkauth
- [x] qt5-purchasing
- [x] qt5-quickcontrols
- [x] qt5-quickcontrols2
- [x] qt5-remoteobjects
- [x] qt5-script
- [x] qt5-scxml
- [x] qt5-sensors
- [x] qt5-serialport
- [x] qt5-speech
- [x] qt5-svg
- [x] qt5-tools
- [x] qt5-virtualkeyboard
- [x] qt5-webchannel
- [x] qt5-websockets
- [x] qt5-webview
- [x] qt5-winextras : only support Windows
- [x] qt5-xmlpatterns

Related: #4560 #6240 #5881 #731 #513.

### CODE REVIEW

**Comment by JackBoosY** [comment]
@Rastaban @ras0219-msft 
Qt5Bootstrap.lib generates **release** library under both **debug** and **release**, so it must be placed under path _tools_, otherwise check will not pass.
However, **qt5-tools** requires Qt5Bootstrap.lib under path _lib_ and path _debug/lib_. Is there any good attention? ?

**Comment by heydojo** [comment]
Building Qt static on Windows in general has been an exercise in futility for a very long while now.
It has never been recommended to build Qt static and in fact those who are, probably don't understand why and what they are doing is completely unnecessary.

> Qt5Bootstrap.lib generates release library under both debug and release, so it must be placed under path tools, otherwise check will not pass.
> However, qt5-tools requires Qt5Bootstrap.lib under path lib and path debug/lib. Is there any good idea?

Yes. vcpkg should by default build only debug or release. Not both. And it must not be possible to build both as you can now. I have no idea why you would need both building at the same time anyway..

So the issue you have here is due to architectural design in vcpkg. The fix is fairly simple. Implementing it might not be.

**Comment by Voskrese** [comment]
@JackBoosY 
the glib library has the same problems, at least if you build through meson
I isolated debug and realise of Assembly, including tools, the last did not begin to do, deleted this folder and placed in BIN

**Comment by JackBoosY** [comment]
@heydojo This is only because qt5 **supports** static builds and some projects (such as Bitcoin) use static libraries.

**Comment by JackBoosY** [comment]
@Voskrese Are you saying [this](https://github.com/microsoft/vcpkg/pull/6883)?

**Comment by Voskrese** [comment]
@JackBoosY 
No, not about that, but it's also a problem for MESON and other assembly systems.
executable files are distinguished by calls to the debug\release libraries

> I hope correctly translates))

**Comment by Voskrese** [comment]
windows static ok
dynamic failed

> jom 1.1.3 - empower your cores
> The system cannot find the path specified.

replaced by nmake ?

I noticed a long time ago that JOM cannot work \ process files in multithreaded mode
either put him flag j1

**Comment by JackBoosY** [comment]
@Voskrese It seems that the file was not created, I don't think this is a jom bug.
I am looking for a reason why it was not created.

**Comment by Voskrese** [comment]
?
`jom.exe -j 12 -f Makefile`

```
set MAKEFLAGS=L
set JOMFLAGS=Lj8
```

**Comment by JackBoosY** [comment]
@Voskrese 
```
PS F:\qt5\vcpkg\buildtrees\qt5-graphicaleffects\x86-windows-dbg> F:/qt5/vcpkg/downloads/tools/jom/jom-1.1.3/jom.exe -j 12 -f Makefile

jom 1.1.3 - empower your cores

        cd src\ && ( if not exist Makefile F:\qt5\vcpkg\installed\x86-windows\tools\qt5\qmake.exe -o Makefile F:\qt5\vcpkg\buildtrees\qt5-graphicaleffects\src\5.12.3-566571ebd9\src\src.pro "CONFIG-=release" "CONFIG+=debug" -qtconf F:/qt5/vcpkg/buildtrees/qt5-graphicaleffects/x86-windows-dbg/qt.conf ) && F:\qt5\vcpkg\downloads\tools\jom\jom-1.1.3\jom.exe -f Makefile
        cd effects\ && ( if not exist Makefile F:\qt5\vcpkg\installed\x86-windows\tools\qt5\qmake.exe -o Makefile F:\qt5\vcpkg\buildtrees\qt5-graphicaleffects\src\5.12.3-566571ebd9\src\effects\effects.pro "CONFIG-=release" "CONFIG+=debug" -qtconf F:/qt5/vcpkg/buildtrees/qt5-graphicaleffects/x86-windows-dbg/qt.conf ) && F:\qt5\vcpkg\downloads\tools\jom\jom-1.1.3\jom.exe -f Makefile
        ( if not exist Makefile.effects_plugin F:\qt5\vcpkg\installed\x86-windows\tools\qt5\qmake.exe -o Makefile.effects_plugin F:\qt5\vcpkg\buildtrees\qt5-graphicaleffects\src\5.12.3-566571ebd9\src\effects\effects_plugin.pro "CONFIG-=release" "CONFIG+=debug" -qtconf F:/qt5/vcpkg/buildtrees/qt5-graphicaleffects/x86-windows-dbg/qt.conf ) && F:\qt5\vcpkg\downloads\tools\jom\jom-1.1.3\jom.exe -f Makefile.effects_plugin
        F:\qt5\vcpkg\downloads\tools\jom\jom-1.1.3\jom.exe -f Makefile.effects_plugin.Debug
        cd private\ && ( if not exist Makefile F:\qt5\vcpkg\installed\x86-windows\tools\qt5\qmake.exe -o Makefile F:\qt5\vcpkg\buildtrees\qt5-graphicaleffects\src\5.12.3-566571ebd9\src\effects\private\private.pro "CONFIG-=release" "CONFIG+=debug" -qtconf F:/qt5/vcpkg/buildtrees/qt5-graphicaleffects/x86-windows-dbg/qt.conf ) && F:\qt5\vcpkg\downloads\tools\jom\jom-1.1.3\jom.exe -f Makefile
        F:\qt5\vcpkg\downloads\tools\jom\jom-1.1.3\jom.exe -f Makefile.Debug
        F:\qt5\vcpkg\installed\x86-windows\tools\qt5-declarative\qmlcachegen.exe -o ..\..\..\qml\QtGraphicalEffects\private\DropShadowBase.qmlc F:\qt5\vcpkg\buildtrees\qt5-graphicaleffects\src\5.12.3-566571ebd9\src\effects\private\DropShadowBase.qml
The system cannot find the path specified.
jom: F:\qt5\vcpkg\buildtrees\qt5-graphicaleffects\x86-windows-dbg\src\effects\private\Makefile.Debug [..\..\..\qml\QtGraphicalEffects\private\DropShadowBase.qmlc] Error 1
        F:\qt5\vcpkg\installed\x86-windows\tools\qt5-declarative\qmlcachegen.exe -o ..\..\..\qml\QtGraphicalEffects\private\FastGlow.qmlc F:\qt5\vcpkg\buildtrees\qt5-graphicaleffects\src\5.12.3-566571ebd9\src\effects\private\FastGlow.qml
        F:\qt5\vcpkg\installed\x86-windows\tools\qt5-declarative\qmlcachegen.exe -o ..\..\..\qml\QtGraphicalEffects\private\FastInnerShadow.qmlc F:\qt5\vcpkg\buildtrees\qt5-graphicaleffects\src\5.12.3-566571ebd9\src\effects\private\FastInnerShadow.qml
The system cannot find the path specified.
jom: F:\qt5\vcpkg\buildtrees\qt5-graphicaleffects\x86-windows-dbg\src\effects\private\Makefile.Debug [..\..\..\qml\QtGraphicalEffects\private\FastGlow.qmlc] Error 1
The system cannot find the path specified.
jom: F:\qt5\vcpkg\buildtrees\qt5-graphicaleffects\x86-windows-dbg\src\effects\private\Makefile.Debug [..\..\..\qml\QtGraphicalEffects\private\FastInnerShadow.qmlc] Error 1
jom: F:\qt5\vcpkg\buildtrees\qt5-graphicaleffects\x86-windows-dbg\src\effects\private\Makefile [debug] Error 2
jom: F:\qt5\vcpkg\buildtrees\qt5-graphicaleffects\x86-windows-dbg\src\effects\Makefile [sub-private-make_first] Error 2
jom: F:\qt5\vcpkg\buildtrees\qt5-graphicaleffects\x86-windows-dbg\src\Makefile [sub-effects-make_first] Error 2
jom: F:\qt5\vcpkg\buildtrees\qt5-graphicaleffects\x86-windows-dbg\Makefile [sub-src-make_first] Error 2
```

File: ports/qt5-base/portfile.cmake [hunk 1]
```diff
@@ -44,7 +44,6 @@ set(CORE_OPTIONS
     -system-zlib
     -system-libjpeg
     -system-libpng
-    -system-freetype
```
Comment (Neumann-A) [inline ports/qt5-base/portfile.cmake]: whats wrong with system freetype?

File: ports/qt5-base/portfile.cmake [hunk 2]
```diff
@@ -203,16 +235,22 @@ if(EXISTS ${CURRENT_PACKAGES_DIR}/lib/qtmain.lib)
     if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "release")
         file(COPY ${CURRENT_PACKAGES_DIR}/lib/qtmain.lib DESTINATION ${CURRENT_PACKAGES_DIR}/lib/manual-link)
         file(COPY ${CURRENT_PACKAGES_DIR}/lib/qtmain.prl DESTINATION ${CURRENT_PACKAGES_DIR}/lib/manual-link)
-        file(REMOVE ${CURRENT_PACKAGES_DIR}/lib/qtmain.lib)
-        file(REMOVE ${CURRENT_PACKAGES_DIR}/lib/qtmain.prl)
+        # qt5-declarative:windows-static need this lib in path lib
```
Comment (Neumann-A) [inline ports/qt5-base/portfile.cmake]: probably related to https://github.com/microsoft/vcpkg/pull/6956#issuecomment-519395463

File: ports/qt5-base/portfile.cmake (hunk 1 above)
Comment (JackBoosY) [reply ports/qt5-base/portfile.cmake]: Related reason:
https://github.com/microsoft/vcpkg/pull/7108#issuecomment-508000562

**Comment by JackBoosY** [comment]
Now the only problem is dynamic build.

File: ports/qt5-base/portfile.cmake (hunk 1 above)
Comment (Neumann-A) [reply ports/qt5-base/portfile.cmake]: -system-prefix is probably fine if you add the bzip dependency to the freetype libs. That is the real reason why qt5-declarative and qt5-ativeqt fail on windows. by using qt-freetype your are just moving that problem aside without solving it :)

**Comment by JackBoosY** [comment]
Since #7667 fixing this issue, close this PR.

**Pull request closed without merging**